pkg/publicapi/middleware/error_handler.go

```diff
@@ -0,0 +1,47 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/bacalhau-project/bacalhau/pkg/publicapi/apimodels"
+	"github.com/labstack/echo/v4"
+	"github.com/rs/zerolog/log"
+)
+
+func CustomHTTPErrorHandler(err error, c echo.Context) {
+
+	var code int
+	var message string
+
+	switch e := err.(type) {
+
+	case *apimodels.APIError:
+		// If it is already our custom APIError, use its code and message
+		code = e.HTTPStatusCode
+		message = e.Message
+
+	default:
+		// In an ideal world this should never happen. We should always have are errors
+		// from server as APIError. If output is this generic string, one should evaluate
+		// and map it to APIError and send in appropriate message.= http.StatusInternalServerError
+		message = "internal server error"
+	}
```

Comment:
as discussed earlier, lets explore appending the actual error message if running in debug mode

Reply:
Done